### PR TITLE
Gopherage command to support Metadata.json

### PR DESCRIPTION
**Pull request by MushuEE**
*somehow really messed up the Branch trying to rebase to remove go.mod changes and it was easier to just redo it

This command will generate metadata about how and where coverage was collected

Other services will use this to properly route and track coverage
/cc @amwat 
/area gopherage

### CODE REVIEW

**Comment by MushuEE** [comment]
Remake of #20959 after git commit nightmare

File: gopherage/cmd/metadata/metadata.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// From https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables
+	base_ref   = "PULL_BASE_REF"
```
Comment (amwat) [inline gopherage/cmd/metadata/metadata.go:33]: this is only available in presubmits

File: gopherage/cmd/metadata/metadata.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to fetch Commit Hash from within covered repo: %v.", err)
+			os.Exit(1)
+		}
+		flags.commitID = commit
+	}
+
+	if flags.ref == "" {
+		ref, err := gitCommand("branch", "--show-current")
+
```
Comment (amwat) [inline gopherage/cmd/metadata/metadata.go:108]: we should still fallback to env if git doesn't work

**Review by amwat — COMMENTED** [review]
/lgtm
/approve
/hold

couple of additions, fine as follow up

File: gopherage/cmd/metadata/metadata.go (hunk 2 above)
Comment (MushuEE) [reply gopherage/cmd/metadata/metadata.go:108]: Done

**Review by amwat — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**